### PR TITLE
Added capability to parse python stack traces

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
Added capability to parse python stack traces

### CODE REVIEW

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs [hunk 1]
```diff
@@ -51,85 +55,121 @@ public override void Parse(LogData logData)
...
                         {
-                            PublishAndReset();
+                            PublishAndReset(logData);
+                            return;
+                        }
+
+                        if (string.IsNullOrWhiteSpace(logData.Line))
+                        {
                             return;
                         }
```
Comment (singhsarab) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs]: Move above

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs [hunk 2]
```diff
@@ -51,85 +55,121 @@ public override void Parse(LogData logData)
...
+                            this.logger.Info($"PythonTestResultParser : ExpectingTestResults: transisitoned to state ExpectingFailedResults at line {logData.LineNumber}");
+
                             return;
                         }
+
                         if (TryParseSummaryTestAndTime(logData))
                         {
                             this.partialTestResult = null;
                             this.state = ParserState.ExpectingSummary;
+                            this.logger.Info($"PythonTestResultParser : ExpectingTestResults: transisitoned to state ExpectingSummary at line {logData.LineNumber}");
```
Comment (singhsarab) [inline Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs]: >transisitoned [](start = 94, length = 13)

nit:

File: Agent.Plugins.UnitTests/PythonTestResultParserTests/Resources/StackTraceTests/TestCase003.txt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+test_skip (__main__.TestStringMethods) ... skipped 'demonstrating skipping'
+test_skip2 (__main__.TestStringMethods) ... skipped 'demonstrating skipping'
+test_split (__main__.TestStringMethods) ... This is output for test_split
+test_split contains multiple lines
+ok
+
+======================================================================
+FAIL: test_fail (__main__.TestStringMethods)
+----------------------------------------------------------------------
+Traceback (most recent call last):
```
Comment (singhsarab) [inline Agent.Plugins.UnitTests/PythonTestResultParserTests/Resources/StackTraceTests/TestCase003.txt]: Add a case when StackTrace is empty.

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs (hunk 1 above)
Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs]: done

File: Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs (hunk 2 above)
Comment (ShreyasRmsft) [reply Agent.Plugins.Log.TestResultParser.Parser/Parser/Python/PythonTestResultParser.cs]: done

**Pull request merged**